ignite/pkg/cosmosgen/templates/rest/rest.ts.tpl

```diff
@@ -122,258 +120,37 @@ export class HttpClient<SecurityDataType = unknown> {
-   */
-  queryDenomMetadataByQueryString = (
-     query?: {
-      "denom": string;
-    },
-    params: RequestParams = {}) =>
-    this.request<QueryDenomMetadataByQueryStringResponse>({
-      path: `/cosmos/bank/v1beta1/denoms_metadata_by_query_string`,
-      method: "GET",
-      query: query,
-      format: "json",
-      ...params,
-    });
-
-  /**
-   * No description
-   *
-   * @tags Query
-   * @name QueryParams
-   * @summary Params queries the parameters of x/bank module.
-   * @request GET:/cosmos/bank/v1beta1/params
-   */
-  queryParams = (params: RequestParams = {}) =>
-    this.request<QueryParamsResponse>({
-      path: `/cosmos/bank/v1beta1/params`,
-      method: "GET",
-      format: "json",
-      ...params,
-    });
-
-  /**
- * @description Since: cosmos-sdk 0.46
- * 
- * @tags Query
- * @name QuerySpendableBalances
- * @summary SpendableBalances queries the spenable balance of all coins for a single
-account.
- * @request GET:/cosmos/bank/v1beta1/spendable_balances/{address}
- */
-  querySpendableBalances = (
-    address: string,
-    query?: {
-      "pagination.key"?: string;
-      "pagination.offset"?: string;
-      "pagination.limit"?: string;
-      "pagination.count_total"?: boolean;
-      "pagination.reverse"?: boolean;
-    },
-    params: RequestParams = {},
-  ) =>
-    this.request<QuerySpendableBalancesResponse>({
-      path: `/cosmos/bank/v1beta1/spendable_balances/${address}`,
-      method: "GET",
-      query: query,
-      format: "json",
-      ...params,
-    });
-
-  /**
-   * No description
+   * {{ .FullName }}
    *
    * @tags Query
-   * @name QueryTotalSupply
-   * @summary TotalSupply queries the total supply of all coins.
-   * @request GET:/cosmos/bank/v1beta1/supply
+   * @name query{{ .Name }}
+   * @request GET:{{ (index .Rules 0).Endpoint }}
    */
-  queryTotalSupply = (
+  query{{ .Name }} = (
+    {{- if (index .Rules 0).Params }}
+    {{- range $i, $param := (index .Rules 0).Params }}{{ if $i }}, {{ end }}{{ $param }}: string{{- end }},
+    {{- end }}{{- if .Paginated }}
     query?: {
       "pagination.key"?: string;
       "pagination.offset"?: string;
       "pagination.limit"?: string;
       "pagination.count_total"?: boolean;
       "pagination.reverse"?: boolean;
-    },
+    },{{ else}}
+    query?: Record<string, any>,{{- end}}
     params: RequestParams = {},
   ) =>
-    this.request<QueryTotalSupplyResponse>({
-      path: `/cosmos/bank/v1beta1/supply`,
-      method: "GET",
-      query: query,
-      format: "json",
-      ...params,
-    });
-
-  /**
-   * No description
-   *
-   * @tags Query
-   * @name QuerySupplyOf
-   * @summary SupplyOf queries the supply of a single coin.
-   * @request GET:/cosmos/bank/v1beta1/supply/by_denom
-   */
-  querySupplyOf = (query?: { denom?: string }, params: RequestParams = {}) =>
-    this.request<QuerySupplyOfResponse>({
-      path: `/cosmos/bank/v1beta1/supply/by_denom`,
+    this.request<Query{{ .Name }}Response>({
```

Comment:
The proto to ts- types conversion using buf seems to switch snakecase to camelcase for properties...this means that the return type here may be wrong

Trying to remember if earlier versions were using snake case or camel case...

Telescope generated types and Cosmjs facour camelCase so Id rather have us be consistent...at the same time I dont want us to redeclare all the types with a different casing format.

Will see if its possible to fix with a generic type